bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -437,12 +437,21 @@ func (r *KubeadmConfigReconciler) handleClusterNotInitialized(ctx context.Contex
 	}
 
 	certificates := secret.NewCertificatesForInitialControlPlane(scope.Config.Spec.ClusterConfiguration)
-	err = certificates.LookupOrGenerate(
-		ctx,
+
+	// Find the certificates associated with the Cluster.
+	err = certificates.Lookup(ctx,
```

Comment:
Let's move this into the else branch of the if

I think this makes.a bit clearer that it's an if/else situation + we avoid a duplicate lookup in the ControlPlaneRef == nil case